cirq/contrib/quirk/url_to_circuit.py

```diff
@@ -13,22 +13,86 @@
 # limitations under the License.
 import json
 import urllib.parse
-from typing import Any, List, Dict, Optional
+from typing import Any, List, Dict, Optional, Sequence, cast, TYPE_CHECKING, \
+    Iterable
 
-import cirq
+from cirq import devices, circuits, ops
 from cirq.contrib.quirk.cells import (
     Cell,
     CellMaker,
     CellMakerArgs,
     generate_all_quirk_cell_makers,
+    ExplicitOperationsCell,
 )
 
-
-def quirk_url_to_circuit(quirk_url: str) -> 'cirq.Circuit':
+if TYPE_CHECKING:
+    import cirq
+
+
+def quirk_url_to_circuit(
+        quirk_url: str,
+        *,
+        qubits: Optional[Sequence['cirq.Qid']] = None,
+        extra_recognized: Iterable['cirq.contrib.quirk.cells.CellMaker'] = ()
+) -> 'cirq.Circuit':
+    """Parses a Cirq circuit out of a Quirk URL.
+
+    Args:
+        quirk_url: The URL of a bookmarked Quirk circuit. It is not required
+            that the domain be "algassert.com/quirk". The only important part of
+            the URL is the fragment (the part after the #).
```

Comment:
Waiting for the day when an RFC defines a URL namespace for compact specification of quantum circuits :-)

Reply:
Acknowledged.